assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -559,6 +561,54 @@ export const DEFAULT_NOTIFICATIONS = {
 			);
 		},
 	},
+	'web-data-stream-not-available-notification': {
+		Component: WebDataStreamNotAvailableNotification,
+		priority: 10,
+		areaSlug: NOTIFICATION_AREAS.BANNERS_ABOVE_NAV,
+		viewContexts: [ VIEW_CONTEXT_MAIN_DASHBOARD ],
+		isDismissible: false,
+		checkRequirements: async ( { select, resolveSelect }, viewContext ) => {
+			const viewOnly =
+				SITE_KIT_VIEW_ONLY_CONTEXTS.includes( viewContext );
+
+			await resolveSelect( CORE_MODULES ).getModules();
+
+			const ga4ModuleConnected = await select(
+				CORE_MODULES
+			).isModuleConnected( 'analytics-4' );
+
+			const hasGTMScope = await select( CORE_USER ).hasScope( GTM_SCOPE );
```

Comment:
You will have to ensure all selectors used in this function are resolved first. So just find out which resolver each selector relies on and call it. Follow the pattern used here:
https://github.com/google/site-kit-wp/blob/819d2a5b4b50c5eabd7929f3478ef4a1d481fd8d/assets/js/googlesitekit/notifications/register-defaults.js#L81-L91

You will need getAuthentication, getModules, getSettings (from analytics-4) and getUser resolvers.